src/codegen/mod.rs

```diff
@@ -3071,23 +3114,25 @@ impl TryToRustTy for Type {
             }
             TypeKind::ResolvedTypeRef(inner) => inner.try_to_rust_ty(ctx, &()),
             TypeKind::TemplateAlias(..) |
-            TypeKind::Alias(..) => {
+            TypeKind::Alias(..) |
+            TypeKind::BlockPointer(..) => {
                 let template_params = item.used_template_params(ctx)
                     .into_iter()
                     .filter(|param| param.is_template_param(ctx, &()))
                     .collect::<Vec<_>>();
 
-                let spelling = self.name().expect("Unnamed alias?");
                 if item.is_opaque(ctx, &()) && !template_params.is_empty() {
                     self.try_to_opaque(ctx, item)
-                } else if let Some(ty) = utils::type_from_named(
-                    ctx,
-                    spelling,
-                )
-                {
-                    Ok(ty)
                 } else {
-                    utils::build_path(item, ctx)
+                    if let Some(spelling) = self.name() {
+                        if let Some(ty) = utils::type_from_named(ctx, spelling) {
```

Comment:
I understand this can be needed, if we share this code between aliases and blocks. Do we have tests for anonymous blocks?

Also, this could use `and_then` to avoid duplicating the `else` branches. Something like:

```rust
} else if let Some(ty) = self.name().and_then(|name| utils::type_from_named(ctx, name)) {
    Ok(ty)
} else {
    utils::build_path(..)
}
```

Though I don't feel terribly strongly about it.

Reply:
I'm not sure what's your meanings about anonymous blocks, but I test two cases, I guess most of function signature will use those.

```objc
void atexit_b(void (^)(void));

typedef void *dispatch_data_t;

typedef bool (^dispatch_data_applier_t)(dispatch_data_t region,
                                        size_t offset,
                                        const void *buffer,
                                        size_t size);

bool dispatch_data_apply(dispatch_data_t data,
                         dispatch_data_applier_t applier);
```